src/libraries/Common/src/System/Security/Cryptography/CryptoThrowHelper.Windows.cs

```diff
@@ -13,19 +17,54 @@ public static CryptographicException ToCryptographicException(this int hr)
         {
             string message = Interop.Kernel32.GetMessage(hr);
 
-            if ((hr & 0x80000000) != 0x80000000)
+            // If the incoming value is non-negative, it's a Win32 error instead of an
+            // HRESULT. We'll convert it to an HRESULT now (subsystem = 0x0007 [win32]).
+            if (hr >= 0)
+            {
                 hr = (hr & 0x0000FFFF) | unchecked((int)0x80070000);
+            }
+
+#if NETCOREAPP3_1_OR_GREATER
+            return new CryptographicException(message)
+            {
+                HResult = hr
+            };
+#else
+            // Prior to .NET Core 3.1, the Exception.HResult property was not publicly
+            // settable, and CryptographicException did not have a ctor which allowed
+            // setting both the message and the HRESULT. We use a subclassed helper
+            // type to allow flowing both pieces of data to receivers.
 
             return new WindowsCryptographicException(hr, message);
```

Comment:
System.Security.Cryptography.Pkcs on the netstandard2.0 build?

Reply:
Sorry - I don't understand the question?

Reply:
I'm trying to understand when we're building where we still hit this code.  System.Security.Cryptography.dll only builds for the current release, so my initial instinct was that the WindowsCryptographicException type was going to be deleted by this change.